### DISCUSSION TITLE
Access to store from .js file

### DISCUSSION THREAD

**Discussion by josejachuf** [quasarframework/quasar#10596] [General - Components / Directives / etc]
Hi, The situation is, I think, similar to this [1]
I'm using version 2 (vue3)

In a file **.vue** I can access the store doing:
```

import { useStore } from 'vuex'
...
const $store = useStore()
$store.dispatch('foo/setBar', data) 
```

I want to be able to do the same but from a **.js** file, but I get **$store is undefined**

What is missing? 

[1] https://github.com/quasarframework/quasar/discussions/10288

**Comment by JeffSpies** [comment]
You'll likely need to import from index.js in quasar's store folder. For example,

```js
import store from './src/store'
const $store = store()
```
Hope that helps!

**Reply by josejachuf** [reply]
Thanks @JeffSpies 
What you told me it worked fine.

The doubt that I have now is why if I call the action from a .vue, by clicking on a button for example, the behavior is reactive, instead with the call from the .js no . 

In this case what I am trying is to modify a title. With the button I see the change instantaneously, with the call from the .js there is no change in the title 

Jose

**Reply by josejachuf** [reply]
Actually what I see that the action if it is executed with the dispatch, calls the mutation, the new value is asiggned to the state, but this is not reflected. I'm looking at it with vue-js-devools and I do not see the change

**Reply by JeffSpies** [reply]
Can you say more about how you're "call[ing it] from .js"?

**Reply by josejachuf** [reply]
Hi @JeffSpies 
This is a simplification of the code

For expample in Index.vue

<script>
import { defineComponent } from 'vue';
import { useStore } from 'vuex'

export default defineComponent({
  name: 'PageIndex',
  components: {
  },
  setup () {

    const $store = useStore()
    const setBar = () => {

      $store.dispatch('foo/setBar', { title: 'Test' })
    }
    return {
      setBar
    }
  }
})
</script>

in button I have @click="setBar"

This works fine. I can see the click the button as in my header the title changed

---------------------------------------

In myfile.js

import store from '../store'

export function setBar() {
  const $store = store()
  $store.dispatch('foo/setBar', { title: 'Test' })
}

When I call setBar I see that the action and the mutation is executed, but I do not see the change of the title reflected

**Reply by metalsadman** [reply]
you likely didn't commit any state within your action, or the `title` you are referring to is not a computed property coming from your vuex state.
